js/ui/marker.js

```diff
@@ -13,6 +13,10 @@ var Point = require('point-geometry');
  * @param {HTMLElement=} element DOM element to use as a marker (creates a div element by default)
  * @param {Object=} options
  * @param {PointLike=} options.offset The offset in pixels as a [`PointLike`](#PointLike) object to apply relative to the element's top left corner. Negatives indicate left and up.
+ * @param {boolean=} [options.pitchScale=false] Whether the marker should scale smaller into the background or larger into the foreground when the map is pitched and rotated
+ * @param {boolean=} [options.zoomScale=false] Whether the marker should scale with the map as you zoom in and out
+ * @param {number=} [options.nativeZoom=map.getZoom()] The native zoom level of the marker. If the map is at this zoom the marker will not be scaled. If zoom is greater the marker will be enlarged. If zoom is less the marker will be shrunk.
+ * @param {number=} [options.zoomFactor=1.6] The multiple by which to scale the size of the marker between each zoom level
```

Comment:
I don't think that works unless you want your native scale to be 10 (which you almost never would). For example, here's the marker.html debug page with those stops:

![image](https://cloud.githubusercontent.com/assets/870796/18531283/24e9fa4e-7a9b-11e6-935f-d75458d90264.png)

I don't really see a good way to add this without completely hacking apart `mapbox-gl-function`

What about just not giving the user an option, getting rid of zoomFactor and just making zoomScale be an integer that represents the native zoom level of the marker.

Eg 

```
var marker = new mapboxgl.Marker(null, {
    zoomScale: 12
});
```

Means "create a marker with a scale of 1 at zoom 12, keep the marker the same size relative to the map as you zoom".